From d68789e71e652db56cfae64b926dbdf09f5e879d Mon Sep 17 00:00:00 2001
Message-Id: <d68789e71e652db56cfae64b926dbdf09f5e879d.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Thu, 15 Jan 2015 20:42:25 +0530
Subject: [PATCH 24/29] MUST_REBASE [VPG]: drm/i915: Add wrapper function to
 get DSI port

This patch adds a wrapper function to get DSI port from I915 driver,
which already parses VBT during its init.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-4868
Change-Id: I8a0682867a913dcd8ff1cd379d754cb4606a8179
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    7 +++++++
 include/drm/i915_adf.h                  |    2 ++
 2 files changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 114a42e..613e39e 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -182,4 +182,11 @@ u8 intel_adf_get_platform_id(void)
 }
 EXPORT_SYMBOL(intel_adf_get_platform_id);
 
+/* To get DSI port no from VBT */
+u16 intel_get_dsi_port_frm_vbt(void)
+{
+	return i915_adf_dev->vbt.dsi.port;
+}
+EXPORT_SYMBOL(intel_get_dsi_port_frm_vbt);
+
 #endif
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index 4349888..fe8cf44 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -57,6 +57,8 @@ extern void intel_adf_display_rpm_get(void);
 extern void intel_adf_display_rpm_put(void);
 extern int i915_adf_driver_initialize(void);
 extern u8 intel_adf_get_platform_id(void);
+extern u16 intel_get_dsi_port_frm_vbt(void);
+
 #endif
 
 
-- 
1.7.9.5

